demo/src/components/BillingInfoForm/BillingInfoForm.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import {Form} from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import {makeFormGroup} from '../UserInfoForm/UserInfoForm.js';
+
+const /** !JSX */ cardType = makeFormGroup(/** label= */ 'Credit Card Type',
+    /** control= */ <Form.Control as="select" custom>
+      <option>Mastercard</option>
+      <option>Visa</option>
+      <option>American Express</option>
+      <option>Discover</option>
+    </Form.Control>,
+);
+
+const /** !JSX */ cardNumber = makeFormGroup(/** label= */ 'Credit Card Type',
+    /** control= */ <Form.Control type='number' required
+      defaultValue='4111111111111111'
+      placeholder='Credit Card Number (required)'
+    />,
+);
+
+const /** !JSX */ zipAndCode = <Form.Row>
+  {makeFormGroup(/** label= */'Security Code',
+      /** control= */ <Form.Control type='number' required
+        defaultValue='111'
+        placeholder='Security Code (required)'
+      />, /** useWholeRow= */ false)}
+  {makeFormGroup(/** label= */ 'Zip Code',
+      /** control= */ <Form.Control
+        type='number'
+        defaultValue='19123'
+        placeholder='Zip (optional)'
+      />, /** useWholeRow= */ false)}
+</Form.Row>;
+
+/**
+ * Creates a page component with a form to get billing information.
+ * @return {!JSX} The component.
+ */
+export function BillingInfoForm({formId}) {
```

Comment:
I see that sometimes methods accept parameter object, sometimes the list of individual parameters, how do you decide where to use each one?

Reply:
The only times that I've done it are when accepting arguments from JSX (like here) or the redux store. Generally speaking, I think it's good to use a parameter object so that every field of the object has a specific individual purpose. With JSX, I could accept a props parameter, but then every time I used said parameter I would be writing props.fieldname, so it's more clear to expand it at first. With the redux store, they always pass in an object with at least 2 fields, representing data and functions to change said data, since these feel like they have 2 different purposes, I split them up.